tests/test_s3.py

```diff
@@ -128,10 +128,25 @@ def test_instantiation(self):
         # attribute on the S3Uploader object.
         eq_('a transform', uploader.url_transform)
 
-    def test_empty_string(self):
+    @parameterized.expand([
+        (
+            'empty_credentials',
+            None,
+            None
+        ),
+        (
+            'empty_string_credentials',
+            '',
+            ''
+        ),
+        (
+            'empty_string_credentials',
```

Comment:
The code change makes sense but I'm still confused about this test name -- the credentials are `username` and `password`, not empty strings. I think you just need to rename this test?

Reply:
I renamed the test to `test_initialization` to reflect that it tests different cases. What do you think would be a good name?

Reply:
test_initialization is fine for the name of the function. I'm talking about the value of the variable `name`. When you're parameterizing the test you set `name="empty_string_credentials"`, `username=""`, `password=""`. That makes sense to me. Then you set `name="empty_string_credentials"` again,  but with `username="username"` and `password="password"`. That's the part I don't get.

Reply:
Sorry, didn't notice that the second case had the same name, changed it to `non_empty_string_credentials`